src/NuGetPackageVerifier/PackageIssueFactory.cs

```diff
@@ -266,12 +266,20 @@ public static PackageVerifierIssue AssemblyHasNoDocFile(string assemblyPath)
             return new PackageVerifierIssue("DOC_MISSING", assemblyPath, string.Format("The assembly '{0}' doesn't have a corresponding XML document file.", assemblyPath), PackageIssueLevel.Warning);
         }
 
-        public static PackageVerifierIssue AssemblyHasWrongJsonNetVersion(string assemblyPath, string targetFramework, string currentVersion, string expectedVersion)
+        public static PackageVerifierIssue PackageHasWrongThirdPartyDependencyVersion(string packageId, string targetFramework, string dependencyPackageId, string currentVersion, string expectedVersion)
         {
             return new PackageVerifierIssue(
-                "WRONG_JSONNET_VERSION",
-                string.Format("{0}; {1}", assemblyPath, targetFramework),
-                string.Format("The assembly '{0}' references the wrong Json.NET version. Current version '{1}'; Expected version '{2}'.", assemblyPath, currentVersion, expectedVersion),
+                "WRONG_THIRDPARTY_DEPENDENCY_VERSION",
+                $"{packageId}; {targetFramework}",
+                $"The package '{packageId}' references the wrong version of `{dependencyPackageId}` package. Current version '{currentVersion}'; Expected version '{expectedVersion}'.",
```

Comment:
Could probably show this wiki page some love https://github.com/aspnet/Coherence-Signed/wiki/NuGetPackageVerifier